src/k8s/pkg/k8sd/setup/kube_apiserver_test.go

```diff
@@ -245,4 +245,30 @@ func TestKubeAPIServer(t *testing.T) {
 		g.Expect(err).To(HaveOccurred())
 		g.Expect(err).To(MatchError(ContainSubstring("unsupported datastore")))
 	})
+
+	t.Run("IPv6", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Create a mock snap
+		s := mustSetupSnapAndDirectories(t, setKubeletMock)
+		s.Mock.Hostname = "dev"
+
+		g.Expect(setup.KubeAPIServer(s, net.ParseIP("2001:db8::"), "fd98::/108", "https://auth-webhook.url", false, types.Datastore{Type: utils.Pointer("k8s-dqlite")}, "Node,RBAC", nil)).To(Succeed())
+
+		tests := []struct {
+			key         string
+			expectedVal string
+		}{
+			{key: "--advertise-address", expectedVal: "2001:db8::"},
```

Comment:
nit: Maybe we can extract these IP strings.

Reply:
this test validates that the `--advertise-address` flag is configured with the IP string that we expect. I don't think it makes sense to have this as an IP object just to stringify it later in the test.

Or do I misunderstand your comment?

Reply:
No no I mean that we could have variables for these IPs that we pass to both `setup.KubeAPIServer()` and `expectedVal`. It's not really important, ignore this if you don't feel like it.